pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -38,7 +53,21 @@ func (pc *PackageControllerClient) InstallController(ctx context.Context) error
 	registry := GetRegistry(pc.uri)
 	sourceRegistry := fmt.Sprintf("sourceRegistry=%s", registry)
 	values := []string{sourceRegistry}
-	return pc.chartInstaller.InstallChart(ctx, pc.chartName, ociUri, pc.chartVersion, pc.kubeConfig, values)
+	err := pc.chartInstaller.InstallChart(ctx, pc.chartName, ociUri, pc.chartVersion, pc.kubeConfig, values)
+	if err != nil {
+		return err
+	}
+
+	err = pc.ApplySecret(ctx)
+	if err != nil {
+		logger.Info("Warning: not able to create secret. Package installation might fail.", "error", err)
```

Comment:
would it make sense to use `logger.Error`?

Reply:
Is this something where if it fails, the user can rerun this command or will it be messy to clean up if package installation fails?

Reply:
The reason it is a warning instead of an error is because there are scenarios where this failure doesn't cause any issue. For example if the user is using registry mirror. We don't want to mention an error in that case. It is also possible the secret was already created before and this error doesn't really impact anything as such.

Reply:
okay as long as even if it fails the cluster state can be recovered somehow

Reply:
This failure doesn't cause any impact. Even install package controller failing doesn't have an impact on the cluster. Otherwise, that would be very concerning.